tests/test_inngest/test_connect/test_lease_extend_failure.py

```diff
@@ -0,0 +1,97 @@
+# pyright: reportPrivateUsage=false
+
+import asyncio
+import typing
+
+import inngest
+import pytest
+import test_core
+from inngest.connect import ConnectionState, connect
+from inngest.connect._internal import connect_pb2
+
+from .base import BaseTest
+
+
+class TestLeaseExtendFailure(BaseTest):
+    @pytest.mark.timeout(10, method="thread")
+    async def test_lease_extend_failure_removes_pending_request(self) -> None:
+        """Test that a lease extension nack removes the pending request."""
+        client = inngest.Inngest(
+            app_id=test_core.random_suffix("app"),
+            is_production=False,
+        )
+        event_name = test_core.random_suffix("event")
+        state = test_core.BaseState()
```

Comment:
We don't need this since we aren't using it in assertions. This is only useful for things like waiting for run status